projects/igniteui-angular/src/lib/toast/toast.component.ts

```diff
@@ -299,12 +341,10 @@ export class IgxToastComponent implements IToggleView, OnInit, OnDestroy {
      * @memberof IgxToastComponent
      */
     public hide(): void {
-        this.onHiding.emit(this);
-        this.isVisible = false;
-        this.onHidden.emit(this);
-        this.animationState = 'invisible';
-
         clearInterval(this.timeoutId);
+        this.isVisible = false;
```

Comment:
at this point toast is still visible. After calling `supper.close()` the first thing that happens close animation start. After close animation ends the content is removed from the DOM and at this point it is not visible anymore. This is why IMO it is better to set `isVisible` in `onClosed` subscription just before we emit `onHidden`.